[Feature Request]: Format Hop Gui Logging output

### What would you like to happen?

Errors can sometimes be hard to spot in the Hop Gui Logging browser because logging output has no fixed format (#5923). An output like this with fixed font  would be favorable:

```
2025/10/28 10:58:41 [BASIC]   Get rows from result.0 - Finished processing (I=0, O=0, R=146, W=292, U=0, E=0)
2025/10/28 11:00:00 [WARNING] Get rows form result.0 - Something terrible happened (I made this up)
2025/10/28 10:58:41 [ERROR]   CSV file input.0 - Unexpected error
2025/10/28 10:58:41 [ERROR]   CSV file input.0 - org.apache.hop.core.exception.HopException: 
2025/10/28 10:58:41 [ERROR]   CSV file input.0 - The filename field 'filename' could not be found.
2025/10/28 10:58:41 [ERROR]   CSV file input.0 - 
2025/10/28 10:58:41 [ERROR]   CSV file input.0 -    at org.apache.hop.pipeline.transforms.csvinput.CsvInput.getFilenamesFromPreviousTransforms(CsvInput.java:276)
2025/10/28 10:58:41 [ERROR]   CSV file input.0 -    at org.apache.hop.pipeline.transforms.csvinput.CsvInput.processRow(CsvInput.java:81)
2025/10/28 10:58:41 [ERROR]   CSV file input.0 -    at org.apache.hop.pipeline.transform.RunThread.run(RunThread.java:54)
2025/10/28 10:58:41 [ERROR]   CSV file input.0 -    at java.base/java.lang.Thread.run(Thread.java:840)
```

Logging output anywhere else should not be altered to ensure that no individual log parsing is affected.

### Issue Priority

Priority: 3

### Issue Component

Component: Hop Gui

--- a/core/src/main/java/org/apache/hop/core/logging/FixedWidthLogLayout.java
+++ b/core/src/main/java/org/apache/hop/core/logging/FixedWidthLogLayout.java
@@ -1,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hop.core.logging;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import org.apache.hop.core.Const;
+
+public class FixedWidthLogLayout {
+  private static final ThreadLocal<SimpleDateFormat> LOCAL_SIMPLE_DATE_PARSER =
+      ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy/MM/dd HH:mm:ss"));
+
+  private static final int MAX_LOG_LEVEL_LENGTH =
+      Arrays.stream(LogLevel.getLogLevelCodes()).map(String::length).reduce(0, Integer::max);
+
+  private boolean timeAdded;
+
+  public FixedWidthLogLayout() {
+    this(true);
+  }
+
+  public FixedWidthLogLayout(boolean addTime) {
+    this.timeAdded = addTime;
+  }
+
+  public String format(HopLoggingEvent event) {
+    StringBuilder line = new StringBuilder();
+
+    String dateTimeString = "";
+
+    if (timeAdded) {
+      dateTimeString = LOCAL_SIMPLE_DATE_PARSER.get().format(new Date(event.timeStamp)) + " ";
+    }
+
+    Object object = event.getMessage();
+
+    if (object instanceof LogMessage message) {
+
+      String logLevel = getLogLevelPadded(message.getLevel());
+
+      String[] parts =
+          message.getMessage() == null ? new String[] {} : message.getMessage().split(Const.CR);
+
+      for (int i = 0; i < parts.length; i++) {
+        if (!message.isSimplified()) {
+          line.append(dateTimeString);
+          line.append(logLevel);
+
+          // Include the subject too on every line...
+          if (message.getSubject() != null) {
+            line.append(message.getSubject());
+            if (message.getCopy() != null) {
+              line.append(".").append(message.getCopy());
+            }
+            line.append(" - ");
+          }
+        }
+
+        line.append(parts[i]);
+        if (i < parts.length - 1) {
+          line.append(Const.CR); // put the CR's back in there!
+        }
+      }
+    } else {
+      line.append(dateTimeString);
+      line.append((object != null ? object.toString() : "<null>"));
+    }
+
+    return line.toString();
+  }
+
+  private String getLogLevelPadded(LogLevel logLevel) {
+    String code = logLevel.getCode();
+    return "[" + code + "] " + " ".repeat(Math.max(MAX_LOG_LEVEL_LENGTH - code.length(), 0));
+  }
+}

--- a/pom.xml
+++ b/pom.xml
@@ -416,9 +416,9 @@
                         <includes>
                             <include>src/**/*.java</include>
                         </includes>
-                        <googleJavaFormat />
-                        <importOrder />
-                        <removeUnusedImports />
+                        <googleJavaFormat></googleJavaFormat>
+                        <importOrder></importOrder>
+                        <removeUnusedImports></removeUnusedImports>
                     </java>
                     <pom>
                         <includes>

--- a/ui/src/main/java/org/apache/hop/ui/hopgui/file/pipeline/HopGuiLogBrowser.java
+++ b/ui/src/main/java/org/apache/hop/ui/hopgui/file/pipeline/HopGuiLogBrowser.java
@@ -28,7 +28,7 @@ import lombok.Getter;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.hop.core.Const;
 import org.apache.hop.core.config.HopConfig;
-import org.apache.hop.core.logging.HopLogLayout;
+import org.apache.hop.core.logging.FixedWidthLogLayout;
 import org.apache.hop.core.logging.HopLogStore;
 import org.apache.hop.core.logging.HopLoggingEvent;
 import org.apache.hop.core.logging.IHasLogChannel;
@@ -76,7 +76,7 @@ public class HopGuiLogBrowser {
     //
     final AtomicInteger lastLogId = new AtomicInteger(-1);
     final AtomicBoolean busy = new AtomicBoolean(false);
-    final HopLogLayout logLayout = new HopLogLayout(true);
+    final FixedWidthLogLayout logLayout = new FixedWidthLogLayout(true);
 
     // Refresh the log every second or so
     //

--- a/ui/src/main/java/org/apache/hop/ui/hopgui/file/pipeline/delegates/HopGuiPipelineLogDelegate.java
+++ b/ui/src/main/java/org/apache/hop/ui/hopgui/file/pipeline/delegates/HopGuiPipelineLogDelegate.java
@@ -38,6 +38,7 @@ import org.apache.hop.ui.hopgui.HopGui;
 import org.apache.hop.ui.hopgui.file.IHopFileTypeHandler;
 import org.apache.hop.ui.hopgui.file.pipeline.HopGuiLogBrowser;
 import org.apache.hop.ui.hopgui.file.pipeline.HopGuiPipelineGraph;
+import org.apache.hop.ui.hopgui.file.shared.TextZoom;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.custom.CTabItem;
 import org.eclipse.swt.layout.FormAttachment;
@@ -58,6 +59,9 @@ public class HopGuiPipelineLogDelegate {
   public static final String TOOLBAR_ICON_LOG_COPY_TO_CLIPBOARD =
       "ToolbarIcon-10020-LogCopyToClipboard";
   public static final String TOOLBAR_ICON_LOG_PAUSE_RESUME = "ToolbarIcon-10030-LogPauseResume";
+  public static final String TOOLBAR_ICON_LOG_INCREASE_FONT = "ToolbarIcon-10040-LogIncreaseFont";
+  public static final String TOOLBAR_ICON_LOG_DECREASE_FONT = "ToolbarIcon-10050-LogDecreaseFont";
+  public static final String TOOLBAR_ICON_LOG_RESET_FONT = "ToolbarIcon-10060-LogResetFont";
 
   private final HopGuiPipelineGraph pipelineGraph;
 
@@ -66,6 +70,7 @@ public class HopGuiPipelineLogDelegate {
   @Getter private CTabItem pipelineLogTab;
 
   private Text pipelineLogText;
+  private TextZoom textZoom;
 
   private ToolBar toolbar;
   private GuiToolbarWidgets toolBarWidgets;
@@ -119,6 +124,10 @@ public class HopGuiPipelineLogDelegate {
     fdText.top = new FormAttachment(toolbar, 0);
     fdText.bottom = new FormAttachment(100, 0);
     pipelineLogText.setLayoutData(fdText);
+
+    this.textZoom = new TextZoom(pipelineLogText, GuiResource.getInstance().getFontFixed());
+    this.textZoom.resetFont();
+
     // add a CR to avoid fontStyle from getting lost on macos HOP-2583
     if (OsHelper.isMac()) {
       pipelineLogText.setText(Const.CR);
@@ -304,6 +313,36 @@ public class HopGuiPipelineLogDelegate {
       item.setImage(GuiResource.getInstance().getImageRun());
       item.setToolTipText(BaseMessages.getString(PKG, "PipelineLog.Dialog.Resume.Tooltip"));
     }
+  }
+
+  @GuiToolbarElement(
+      root = GUI_PLUGIN_TOOLBAR_PARENT_ID,
+      id = TOOLBAR_ICON_LOG_INCREASE_FONT,
+      toolTip = "i18n:org.apache.hop.ui.hopgui:WorkflowLog.Button.IncreaseFont",
+      image = "ui/images/zoom-in.svg",
+      separator = true)
+  public void increaseFont() {
+    this.textZoom.increaseFont();
+  }
+
+  @GuiToolbarElement(
+      root = GUI_PLUGIN_TOOLBAR_PARENT_ID,
+      id = TOOLBAR_ICON_LOG_DECREASE_FONT,
+      toolTip = "i18n:org.apache.hop.ui.hopgui:WorkflowLog.Button.DecreaseFont",
+      image = "ui/images/zoom-out.svg",
+      separator = false)
+  public void decreaseFont() {
+    this.textZoom.decreaseFont();
+  }
+
+  @GuiToolbarElement(
+      root = GUI_PLUGIN_TOOLBAR_PARENT_ID,
+      id = TOOLBAR_ICON_LOG_RESET_FONT,
+      toolTip = "i18n:org.apache.hop.ui.hopgui:WorkflowLog.Button.ResetFont",
+      image = "ui/images/zoom-100.svg",
+      separator = false)
+  public void resetFont() {
+    this.textZoom.resetFont();
   }
 
   public boolean hasSelectedText() {

--- a/ui/src/main/java/org/apache/hop/ui/hopgui/file/shared/TextZoom.java
+++ b/ui/src/main/java/org/apache/hop/ui/hopgui/file/shared/TextZoom.java
@@ -1,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hop.ui.hopgui.file.shared;
+
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.FontData;
+import org.eclipse.swt.widgets.Text;
+
+public class TextZoom {
+  public static final int STEP_DEFAULT = 1;
+  public static final int MIN_HEIGHT = 4;
+
+  private Text widget;
+  private Font font;
+  private int step;
+  private int minHeight;
+
+  public TextZoom(Text widget, Font font) {
+    this(widget, font, TextZoom.STEP_DEFAULT);
+  }
+
+  public TextZoom(Text widget, Font font, int step) {
+    this(widget, font, step, MIN_HEIGHT);
+  }
+
+  public TextZoom(Text widget, Font font, int step, int minHeight) {
+    this.widget = widget;
+    this.font = font;
+    this.step = step;
+    this.minHeight = minHeight;
+  }
+
+  public void increaseFont() {
+    increaseHeightBy(step);
+  }
+
+  public void decreaseFont() {
+    increaseHeightBy(-step);
+  }
+
+  public void resetFont() {
+    widget.setFont(font);
+  }
+
+  private void increaseHeightBy(int points) {
+    FontData fontData = widget.getFont().getFontData()[0];
+    int newHeight = Math.max(minHeight, fontData.getHeight() + points);
+    fontData.setHeight(newHeight);
+    widget.setFont(new Font(widget.getDisplay(), fontData));
+  }
+}

--- a/ui/src/main/java/org/apache/hop/ui/hopgui/file/workflow/delegates/HopGuiWorkflowLogDelegate.java
+++ b/ui/src/main/java/org/apache/hop/ui/hopgui/file/workflow/delegates/HopGuiWorkflowLogDelegate.java
@@ -33,6 +33,7 @@ import org.apache.hop.ui.core.widget.OsHelper;
 import org.apache.hop.ui.hopgui.HopGui;
 import org.apache.hop.ui.hopgui.file.IHopFileTypeHandler;
 import org.apache.hop.ui.hopgui.file.pipeline.HopGuiLogBrowser;
+import org.apache.hop.ui.hopgui.file.shared.TextZoom;
 import org.apache.hop.ui.hopgui.file.workflow.HopGuiWorkflowGraph;
 import org.apache.hop.workflow.WorkflowMeta;
 import org.apache.hop.workflow.action.ActionMeta;
@@ -57,6 +58,9 @@ public class HopGuiWorkflowLogDelegate {
   public static final String TOOLBAR_ICON_LOG_COPY_TO_CLIPBOARD =
       "ToolbarIcon-10020-LogCopyToClipboard";
   public static final String TOOLBAR_ICON_LOG_PAUSE_RESUME = "ToolbarIcon-10030-LogPauseResume";
+  public static final String TOOLBAR_ICON_LOG_INCREASE_FONT = "ToolbarIcon-10040-LogIncreaseFont";
+  public static final String TOOLBAR_ICON_LOG_DECREASE_FONT = "ToolbarIcon-10050-LogDecreaseFont";
+  public static final String TOOLBAR_ICON_LOG_RESET_FONT = "ToolbarIcon-10060-LogResetFont";
 
   private HopGui hopGui;
   private HopGuiWorkflowGraph workflowGraph;
@@ -64,6 +68,7 @@ public class HopGuiWorkflowLogDelegate {
   private CTabItem workflowLogTab;
 
   private Text workflowLogText;
+  private TextZoom textZoom;
 
   /** The number of lines in the log tab */
   private Composite workflowLogComposite;
@@ -121,6 +126,10 @@ public class HopGuiWorkflowLogDelegate {
     fdText.top = new FormAttachment((Control) toolbar, 0);
     fdText.bottom = new FormAttachment(100, 0);
     workflowLogText.setLayoutData(fdText);
+
+    this.textZoom = new TextZoom(workflowLogText, GuiResource.getInstance().getFontFixed());
+    this.textZoom.resetFont();
+
     // add a CR to avoid fontStyle from getting lost on macos HOP-2583
     if (OsHelper.isMac()) {
       workflowLogText.setText(Const.CR);
@@ -282,6 +291,36 @@ public class HopGuiWorkflowLogDelegate {
       logBrowser.setPaused(true);
       item.setImage(GuiResource.getInstance().getImagePause());
     }
+  }
+
+  @GuiToolbarElement(
+      root = GUI_PLUGIN_TOOLBAR_PARENT_ID,
+      id = TOOLBAR_ICON_LOG_INCREASE_FONT,
+      toolTip = "i18n:org.apache.hop.ui.hopgui:WorkflowLog.Button.IncreaseFont",
+      image = "ui/images/zoom-in.svg",
+      separator = true)
+  public void increaseFont() {
+    this.textZoom.increaseFont();
+  }
+
+  @GuiToolbarElement(
+      root = GUI_PLUGIN_TOOLBAR_PARENT_ID,
+      id = TOOLBAR_ICON_LOG_DECREASE_FONT,
+      toolTip = "i18n:org.apache.hop.ui.hopgui:WorkflowLog.Button.DecreaseFont",
+      image = "ui/images/zoom-out.svg",
+      separator = false)
+  public void decreaseFont() {
+    this.textZoom.decreaseFont();
+  }
+
+  @GuiToolbarElement(
+      root = GUI_PLUGIN_TOOLBAR_PARENT_ID,
+      id = TOOLBAR_ICON_LOG_RESET_FONT,
+      toolTip = "i18n:org.apache.hop.ui.hopgui:WorkflowLog.Button.ResetFont",
+      image = "ui/images/zoom-100.svg",
+      separator = false)
+  public void resetFont() {
+    this.textZoom.resetFont();
   }
 
   public boolean hasSelectedText() {

--- a/ui/src/main/resources/org/apache/hop/ui/hopgui/messages/messages_de_DE.properties
+++ b/ui/src/main/resources/org/apache/hop/ui/hopgui/messages/messages_de_DE.properties
@@ -14,7 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
 HopGui.Application.Name=Hop
 HopGui.Dialog.ActionCanNotBeChanged.Message=Die Workflow-Action kann nicht ge\u00E4ndert werden!
 HopGui.Dialog.ActionCanNotBeChanged.Title=Sorry...
@@ -241,6 +240,9 @@ WorkflowGraph.Dialog.HopCausesLoop.Message=Dieser Hop verursacht eine unerlaubte
 WorkflowGraph.Dialog.HopCausesLoop.Title=Endlosschleife!
 WorkflowLog.Button.LogSettings=Log Einstellungen
 WorkflowLog.Button.Pause=Workflow Log pausieren
+WorkflowLog.Button.IncreaseFont=Schrift vergr\u00F6\u00DFern
+WorkflowLog.Button.DecreaseFont=Schrift verkleinern
+WorkflowLog.Button.ResetFont=Schriftgr\u00F6\u00DFe zur\u00FCcksetzen
 WorkflowLog.Button.ShowErrorLines=Fehlerzeilen anzeigen
 WorkflowLog.Dialog.SaveJobBeforeRunning.Message=Den Workflow speichern, bevor er gestartet wird.
 WorkflowLog.Dialog.SaveJobBeforeRunning.Title=Workflow speichern

--- a/ui/src/main/resources/org/apache/hop/ui/hopgui/messages/messages_en_US.properties
+++ b/ui/src/main/resources/org/apache/hop/ui/hopgui/messages/messages_en_US.properties
@@ -14,8 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-
 HopGui.Application.Name=Hop
 HopGui.Dialog.ActionCanNotBeChanged.Message=This action can''t be changed\!
 HopGui.Dialog.ActionCanNotBeChanged.Title=Sorry...
@@ -271,6 +269,9 @@ WorkflowLog.Button.ClearLog=Clear log
 WorkflowLog.Button.LogCopyToClipboard=Copy workflow log to clipboard
 WorkflowLog.Button.LogSettings=Log settings
 WorkflowLog.Button.Pause=Pause workflow log
+WorkflowLog.Button.IncreaseFont=Increase font size
+WorkflowLog.Button.DecreaseFont=Decrease font size
+WorkflowLog.Button.ResetFont=Reset font size
 WorkflowLog.Button.ShowErrorLines=Show error lines
 WorkflowLog.Dialog.SaveJobBeforeRunning.Message=Please save the workflow before running it.
 WorkflowLog.Dialog.SaveJobBeforeRunning.Title=Save workflow

